Gameplay/Player.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+#include "../CSC8503/CSC8503Common/GameObject.h"
+#include "Ability.h"
+#include "ePlayerRole.h"
+
+namespace NCL {
+	namespace CSC8503 {
+		class Player : GameObject {
+		public:
+			Player(PlayerRole colour, Vector3 transform);
+			~Player();
+
+			void Move();
+			void Jump();
+			void Shoot();
+			void AssignRole();
+
+		private:
+
+			float health = 100.0f;
+			float speed = 5.0f;
+			
+			// t is short for timer (cooldowns)
+			float tDash;
```

Comment:
Wouldn't we need both a timer for the cooldown, and some kind of variable for the cooldown duration?

Reply:
Yes, we would!